packages/core/src/interpreter.ts

```diff
@@ -417,6 +426,29 @@ export class Interpreter<
     this.doneListeners.add(listener);
     return this;
   }
+  /**
+   * Adds an error listener that is notified when an unhandled error occurs.
+   * @param errorListener The error listener
+   */
+  public onError(
+    errorListener: ErrorListener
+  ): Interpreter<TContext, TStateSchema, TEvent, TTypestate> {
+    this.errorListeners.add(errorListener);
+    return this;
+  }
+
+  private handleError(errorData: unknown): void {
```

Comment:
there are only 2 places in which this is currently called:
- when an error gets caught when we send an error event to itself after catching promise rejection
- when an action throws during `exec`

There are still a lot of uncovered situations:
- errors thrown during guards evaluation
- errors thrown during service instantiation
- errors received from machines (should we hook into their `onError`?) and from observables

Reply:
Right, we should handle errors there as well.